manual/coding_standards.tex

```diff
@@ -0,0 +1,499 @@
+
+\subsubsection{Binary Operators}
+The assignment operator should always have spaces around it.
+Other operators may have spaces around them, but it is not mandatory.
+
+\subsubsection{Unary Operators}
+Do not put any space between an unary operator and their argument.
+
+\subsubsection{Types}
+The \inlinecode{using} syntax is preferred to \inlinecode{typedef} for type aliases.
+If the actual type is not excessively long or complex simple just use it.
+
+The angle brackets of templates should not have any external and internal padding.
+Pointer or Reference operators should go with the type.
+Examples:
+\begin{lstlisting}
+Type* var;
+Type& var;
+
+Class1<Class2<type1>> object;
+\end{lstlisting}
+
+\subsection{Vertical Spacing}
+Use empty lines when it helps to improve the readability of the code, but do not use too many.
+Do not use empty lines after a brace which opens a scope,
+or before a brace which closes a scope.
+Each file should contain an empty line at the end of the file.
+Some editors add an empty line automatically, some do not.
+
+\subsection{Indentation}
+Indentation consists of 2 spaces.
+Do not use tabs in the code.
+
+\subsection{Variable Declarations and Definitions}
+
+\begin{itemize}
+\item Avoid declaring multiple variables in the same declaration, especially if they are not fundamental types:
+
+\begin{lstlisting}[showspaces=false]
+int x, y;                        // Not recommended.
+Matrix a("my-matrix"), b(size);  // Disallowed.
+
+// Preferred way.
+int x;
+int y;
+Matrix a("my-matrix");
+Matrix b(10);
+\end{lstlisting}
+
+\item Use the following order for keywords and modifiers in  variable declarations:
+
+\begin{lstlisting}[showspaces=false]
+// General type
+[static] [const/constexpr] Type variable_name;
+
+// Pointer
+[static] [const] Type* [const] variable_name;
+
+// Integer
+// int is optional, if a signedness or size modifier is present.
+[static] [const/constexpr] [signedness] [size] int variable_name;
+
+// Examples:
+static const Matrix a(10);
+const double* const d(3.14);
+constexpr unsigned long l(42);
+\end{lstlisting}
+
+\end{itemize}
+
+\subsection{Function Declarations and Definitions}
+
+The return type should be on the same line as the function name.
+Parameters should be on the same line, too, unless they do not fit on it.
+Include the parameter names also in the declaration of a function, i.e.
+\begin{lstlisting}
+/** calculates a*b+c
+ */
+double function(double a, double b, double c);
+\end{lstlisting}
+avoid
+\begin{lstlisting}
+Type function(Type1, Type2, Type3);
+\end{lstlisting}
+
+\subsection{Conditionals}
+
+Examples:
+\begin{lstlisting}
+if (condition)
+  statement;
+else
+  statement;
+
+if (condition)
+{
+  statement;
+}
+else if (condition2)
+{
+  statement;
+}
+else
+{
+  statement;
+}
+\end{lstlisting}
+
+\subsection{Switch statement}
+
+Switch statements should always have a default case.
+
+Example:
+\begin{lstlisting}
+switch (var) {
```

Comment:
Clang format will render the open curly brace on the line following switch, right?

Reply:
Yes